src/rules/ResourceICUPluralTranslation.js

```diff
@@ -0,0 +1,249 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IntlMessageFormat } from 'intl-messageformat';
+import Locale from 'ilib-locale';
+import { Rule, Result } from 'i18nlint-common';
+
+/**
+ * @class Represent an ilib-lint rule.
+ */
+class ResourceICUPluralTranslation extends Rule {
+    /**
+     * Make a new rule instance.
+     * @constructor
+     */
+    constructor(options) {
+        super(options);
+        this.name = "resource-icu-plurals-translated";
+        this.description = "Ensure that plurals in translated resources are also translated";
+        this.sourceLocale = (options && options.sourceLocale) || "en-US";
+        this.link = "https://gihub.com/ilib-js/i18nlint/blob/main/docs/resource-icu-plurals-translated.md";
+    }
+
+    getRuleType() {
+        return "resource";
+    }
+
+    /**
+     * Given some ast nodes, reconstruct the string that it came from, glossing
+     * over the plurals and selects, which will be compared separately anyways.
+     * @private
+     */
+    reconstruct(nodes) {
+        let result = "";
+
+        for (let i = 0; i < nodes.length; i++) {
+            switch (nodes[i].type) {
+                case 1:
+                    result += ` {${nodes[i].value}} `;
+                    break;
+
+                case 2:
+                    result += ` {${nodes[i].value}, number, ${nodes[i].style}} `;
+                    break;
+
+                case 3:
+                    result += ` {${nodes[i].value}, date, ${nodes[i].style}} `;
+                    break;
+
+                case 4:
+                    result += ` {${nodes[i].value}, time, ${nodes[i].style}} `;
+                    break;
+
+                // for select ordinals and plurals, replace them with a fixed string
+                // so that we don't match on the details of those subparts
+                case 5:
+                    result += ` {selectordinal} `;
+                    break;
+
+                case 6:
+                    result += ` {plural} `;
+                    break;
+
+                case 7:
+                    result += " # ";
+                    break;
+
+                case 8:
+                    result += `<${nodes[i].value}/>`;
+                    break;
+
+                default:
+                    result += nodes[i].value;
+                    break;
+            }
+        }
+
+        return result.replace(/\s+/g, " ").trim();
+    }
```

Comment:
Actually, we still do need it. Here's the example:

```
source:   {count, plural, one {singular} other {plural {count2, plural, one {1st word} other {other word}}}}
target: {count, plural, one {translated singular} other {plural{count2, plural, one {1st word  } other {other    word}}}}
```

Without adding the space and then collapsing multiple spaces, the reconstructed value of the outer "other" would be different between source and target, even though it is the same text in English. In order to normalize out all the spaces, we add spaces knowing that we may add too many, and then collapse multiple spaces into 1. Also, it would collapse \n -> " " as well. I have definitely seen spacing differences in the stuff coming back from Lilt where the target was not translated.

Reply:
Ok, I mostly agree. However, if we want to treat cases like `"plural {plural}" -> "plural{plural}"` as equal, then maybe we should just remove ALL whitespaces during comparison instead? Because to me, this is a characteristic of how we do _comparison_, not how we intended to do reconstruction (i.e. replacing plurals with placeholders). These reconstructed strings ARE different (`"plural 1st word" -> "plural1st word  "`), it's just we choose to ignore these differences when comparing them.

Reply:
Also, catching this missing space should be a responsibility of a rule like `ResourceEdgeWhitespace` but for plural messages

Reply:
Edge Whitespace matches whitespace, and only at the leading and trailing end of the string. In this case, we want to normalize whitespace, and do that in the middle of the string.

Reply:
Can You please add an explanation for space stuffing as a comment?